core/src/main/java/org/elasticsearch/index/query/SimpleQueryParser.java

```diff
@@ -202,51 +203,135 @@ private Query newPossiblyAnalyzedQuery(String field, String termStr) {
             return new PrefixQuery(new Term(field, termStr));
         }
     }
-
+    
     /**
      * Class encapsulating the settings for the SimpleQueryString query, with
      * their default values
      */
     public static class Settings {
-        private Locale locale = Locale.ROOT;
-        private boolean lowercaseExpandedTerms = true;
-        private boolean lenient = false;
-        private boolean analyzeWildcard = false;
+        /** Locale to use for parsing. */
+        private static final Locale DEFAULT_LOCALE = Locale.ROOT;
+        private Locale locale = DEFAULT_LOCALE;
+
+        /** Specifies whether parsed terms should be lowercased. */
+        private static final boolean DEFAULT_LOWERCASE_EXPANDED_TERMS = true;
+        private Boolean lowercaseExpandedTerms = DEFAULT_LOWERCASE_EXPANDED_TERMS;
+
+        /** Specifies whether lenient query parsing should be used. */
+        private static final boolean DEFAULT_LENIENT = false;
+        private Boolean lenient = DEFAULT_LENIENT;
+
+        /** Specifies whether wildcards should be analyzed. */
+        private static final boolean DEFAULT_ANALYZE_WILDCARD = false;
+        private Boolean analyzeWildcard = DEFAULT_ANALYZE_WILDCARD;
 
+        /**
+         * Generates default {@link Settings} object (uses ROOT locale, does
+         * lowercase terms, no lenient parsing, no wildcard analysis).
+         * */
         public Settings() {
+            //
+        }
 
+        public Settings(Locale locale, Boolean lowercaseExpandedTerms, Boolean lenient, Boolean analyzeWildcard) {
+            this.locale = locale;
+            this.lowercaseExpandedTerms = lowercaseExpandedTerms;
+            this.lenient = lenient;
+            this.analyzeWildcard = analyzeWildcard;
         }
 
-        public void locale(Locale locale) {
+        /** Specifies the locale to use for parsing, Locale.ROOT by default. */
+        public Settings locale(Locale locale) {
             this.locale = locale;
+            return this;
         }
 
+        /** Returns the locale to use for parsing. */
         public Locale locale() {
             return this.locale;
         }
 
-        public void lowercaseExpandedTerms(boolean lowercaseExpandedTerms) {
+        /**
+         * Specifies whether to lowercase parse terms, defaults to true if
+         * unset.
+         */
+        public Settings lowercaseExpandedTerms(Boolean lowercaseExpandedTerms) {
             this.lowercaseExpandedTerms = lowercaseExpandedTerms;
+            return this;
         }
 
-        public boolean lowercaseExpandedTerms() {
+        /** Returns whether to lowercase parse terms. */
+        public Boolean lowercaseExpandedTerms() {
             return this.lowercaseExpandedTerms;
         }
 
-        public void lenient(boolean lenient) {
+        /** Specifies whether to use lenient parsing, defaults to false. */
+        public Settings lenient(Boolean lenient) {
```

Comment:
Good thought - added this only for consistency with the other classes I saw so far but happy to keep void.